Fix don't allow the domain name when creating the cluster

MateSousa opened the pull request:
This closes #96

MateSousa committed:
🚀 feat(network.go): add IsValidDNS function to validate DNS names
The IsValidDNS function is added to the network utility package. This function checks if a given string is a valid DNS name. It verifies that the length of the string is not greater than 255 characters and that each label (separated by '.') is not longer than 63 characters. It also checks that the last character of each label is not a hyphen ('-'). This function can be used to validate DNS names before using them in network-related operations.

MateSousa committed:
🐛 fix(cluster.go): add validation for DNS format in node addresses
The validation for node addresses in the CreateClusterRequest struct has been updated to check for both IP:Port format and valid DNS format. This ensures that only valid node addresses are accepted and prevents invalid addresses from being added to the list of invalidNodes.

git-hulk commented:
Hi @MateSousa 

Thanks for your contribution, would you mind transferring your contribute codes to the Apache group? If yes, please help to reply at https://github.com/RocksLabs/kvrocks_controller/issues/94 before merging.

git-hulk commented on util/network.go:
```diff
@@ -45,3 +45,18 @@ func IsPort(port string) bool {
 	return p > 0 && p < 65536
 }
+
+func IsValidDNS(s string) bool {
```
IsValidDNS => IsDomainName

git-hulk commented on server/cluster.go:
```diff
@@ -54,7 +54,7 @@ func (req *CreateClusterRequest) validate() error {
 	invalidNodes := make([]string, 0)
 	for _, node := range req.Nodes {
-		if !util.IsIPPort(node) {
+		if !util.IsIPPort(node) && !util.IsValidDNS(node) {
```
How about renaming the IsIPPort to IsHostPort, then checking if the host is an IP or a valid domain name? For checking the domain name, it'd be better to use `net.LookupHost` directly.

MateSousa replied:
Great ! did it now

MateSousa committed:
🔀 refactor(network.go): rename IsIPPort function to IsHostPort
🔀 refactor(network.go): add IsDomain function to check if a string is a valid domain
The IsIPPort function has been renamed to IsHostPort to provide a more accurate description of its functionality. Additionally, a new function called IsDomain has been added to check if a string is a valid domain by performing a DNS lookup. This improves the clarity and readability of the code.

MateSousa committed:
🐛 fix(cluster.go): update validation logic for node addresses
The validation logic for node addresses has been updated to use the `util.IsHostPort` function instead of checking for IP:Port and DNS formats separately. This simplifies the code and ensures that only valid host:port combinations are considered valid nodes.

git-hulk commented on util/network.go:
```diff
@@ -26,12 +26,13 @@ import (
 	}
-	return IsIP(parts[0]) && IsPort(parts[1])
+
+	return IsIP(parts[0]) && IsPort(parts[1]) || IsDomain(parts[0]) && IsPort(parts[1])
```
```suggestion
	return (IsIP(parts[0]) || IsDomain(parts[0])) && IsPort(parts[1])
```

git-hulk committed:
Update util/network.go
